src/PrettierEditService.ts

```diff
@@ -465,7 +465,12 @@ export default class PrettierEditService implements Disposable {
     this.loggingService.logInfo("Prettier Options:", prettierOptions);
 
     try {
-      const formattedText = prettierInstance.format(text, prettierOptions);
+      const formattedResult = prettierInstance.format(text, prettierOptions);
+      // Since Prettier v3, `format` returns Promise.
+      const formattedText =
+        typeof formattedResult === "string"
+          ? formattedResult
+          : await formattedResult;
```

Comment:
`prettier.format` now returns a `Promise`, so await dynamically.

Reply:
Actually, this is an async function, so the `Promise` will be resolved by the caller without awaiting here, but await for just in case.

Reply:
string can be awaited too, no need check type.